Detectors/GRP/calibration/include/GRPCalibration/GRPDCSDPsProcessor.h

```diff
@@ -41,25 +41,24 @@ inline unsigned long llu2lu(std::uint64_t v) { return (unsigned long)v; }
 
 struct GRPEnvVariables {
 
-  std::pair<uint64_t, double> mCavernTemperature;
-  std::pair<uint64_t, double> mCavernAtmosPressure;
-  std::pair<uint64_t, double> mSurfaceAtmosPressure;
-  std::pair<uint64_t, double> mCavernAtmosPressure2;
+  std::unordered_map<std::string, std::vector<std::pair<uint64_t, double>>> mEnvVars;
 
   GRPEnvVariables()
   {
-    mCavernTemperature = std::make_pair(0, -999999999);
-    mCavernAtmosPressure = std::make_pair(0, -999999999);
-    mSurfaceAtmosPressure = std::make_pair(0, -999999999);
-    mCavernAtmosPressure2 = std::make_pair(0, -999999999);
+    mEnvVars["CavernTemperature"].emplace_back(0, -999999999);
+    mEnvVars["CavernAtmosPressure"].emplace_back(0, -999999999);
+    mEnvVars["SurfaceAtmosPressure"].emplace_back(0, -999999999);
+    mEnvVars["CavernAtmosPressure2"].emplace_back(0, -999999999);
```

Comment:
why do you need this dummy element? If just to have something to pass to the `processPairD` you can simply pass the vector itself instead of `back()` and always update if it is empty.

Not urgent, but we would need also some easy method to get the value of the point for given timestamp, given that now we store a vector.

Reply:
I wanted to avoid to have sometimes something empty (even if it should not happen, since the first processing will be on a FBI). So I used a dummy first value. I can surely remove it.

Reply:
What is wrong with having empty entry?

Reply:
It was also to initialize the items of the map. Now it is done in the code with the access to the items themselves, l.147:

```
  auto& vect = mapToUpdate[alias]; // we also create at this point the vector in the map if it did not exist yet
```